File: backend/src/main/java/com/simats/neoomfs/dto/response/PatientResponse.java
package com.simats.neoomfs.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PatientResponse {
    private Long id;
    private String mrn;
    private String fullName;
    private Integer age;
    private String gender;
    private LocalDate dateOfBirth;
    private String phoneNumber;
    private String address;
    private String bloodGroup;
    private String emergencyContact;
    private String emergencyPhone;
    private String assessmentStatus;
    private String procedureType;
    private String referringDoctor;
    private String createdByName;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/app/src/main/java/com/simats/neoomfs/models/PatientResponse.kt
package com.simats.neoomfs.models

data class PatientResponse(
    val id: Long,
    val mrn: String?,
    val fullName: String,
    val age: Int?,
    val gender: String?,
    val dateOfBirth: String?,
    val phoneNumber: String?,
    val address: String?,
    val bloodGroup: String?,
    val emergencyContact: String?,
    val emergencyPhone: String?,
    val assessmentStatus: String?,
    val procedureType: String?,
    val referringDoctor: String?,
    val createdByName: String?,
    val createdAt: String?,
    val updatedAt: String?,
    val submittedBy: String? = null,
    val reviewedByName: String? = null,
    val reviewComments: String? = null,
    val approvedAt: String? = null
)
